[UXIT-1397] Zod Ecosystem Explorer Projects

mirhamasala opened the pull request:
## 📝 Description

Please include a summary of the changes. Provide context and motivation for the change, and describe what problem it solves.

- **Type:** Bug fix / New feature / Documentation / Refactor

## 🛠️ Key Changes

- [Change 1 - Brief description]
- [Change 2 - Brief description]
- ...

## 📌 To-Do Before Merging

- [ ] [Task 1 - Brief description]
- [ ] [Task 2 - Brief description]
- ...

## 🧪 How to Test

- **Setup:** [Setup details]
- **Steps to Test:**
  1. [Step 1 - Brief description]
  2. [Step 2 - Brief description]
  3. ...
- **Expected Results:** [Outcome]
- **Additional Notes:** [Additional info]

## 📸 Screenshots

Attach if there are UI changes.

## 🔖 Resources

Feel free to share any references to documentation, libraries, blog posts, or other resources that you consulted or used during the implementation of the changes.

- [Resource 1]
- [Resource 2]
- ...

mirhamasala commented on src/app/_utils/convertMarkdownToEcosystemProjectData.ts:
```diff
@@ -0,0 +1,44 @@
+) {
+  const result = EcosystemProjectFrontMatter.safeParse({
+    title: data.title,
+    slug: data.slug,
```
```suggestion
```

CharlyMartin commented on src/app/events/[slug]/utils/generateStructuredData.ts:
```diff
@@ -1,6 +1,11 @@
-import { Event, Place, VirtualLocation, WithContext } from 'schema-dts'
+import {
+  Event as SchemaEvent,
+  Place,
```
EventSchema maybe?

```suggestion
  Event as EventSchema,
```

CharlyMartin commented on src/app/events/[slug]/utils/generateStructuredData.ts:
```diff
@@ -40,7 +45,7 @@ function getLocation({
 
-export function generateStructuredData(data: EventData): WithContext<Event> {
+export function generateStructuredData(data: Event): WithContext<SchemaEvent> {
   const {
```
```suggestion
export function generateStructuredData(data: Event): WithContext<EventSchema> {
```

CharlyMartin commented on src/content/ecosystem-explorer/projects/dcent.md:
```diff
@@ -16,14 +16,8 @@ tech:
-video-url:
-year-joined: 2023-08-01T20:34:10.316000Z
-news-update:
+twitter: https://twitter.com/Dcent_Web3 2023-08-01T20:34:10.316000Z
```
Something weird is happening here. `2023-08-01T20:34:10.316000Z` is the `year-joinded` date?

CharlyMartin commented on src/content/ecosystem-explorer/projects/filezilla.md:
```diff
@@ -14,14 +14,8 @@ tech:
-video-url:
-year-joined: 2024-02-21T15:20:11.125000Z
-news-update:
+twitter: https://twitter.com/FileZillaToken 2024-02-21T15:20:11.125000Z
```
Same here

CharlyMartin commented on src/content/ecosystem-explorer/projects/flame-launch.md:
```diff
@@ -11,14 +11,8 @@ tech:
-video-url:
-year-joined: 2022-12-01T17:27:13.497000Z
-news-update:
+twitter: https://twitter.com/flamelaunch 2022-12-01T17:27:13.497000Z
```
And here 🙃

CharlyMartin commented on src/content/ecosystem-explorer/projects/pyth-network.md:
```diff
@@ -16,9 +15,7 @@ description: First-party financial data oracle network that delivers low-latency
-repo: https://github.com/pyth-network
-news-update: null
-featured-content: null
+repo: https://github.com/pyth-network null
```
Same problem but with `null` instead of a date

CharlyMartin commented on src/content/ecosystem-explorer/projects/secured-finance.md:
```diff
@@ -11,16 +11,8 @@ tech:
-news-update:
-subcategories:
-  - finance
+twitter: https://twitter.com/Secured_Fi 2024-03-29T21:28:54.282000Z
```
Here too

CharlyMartin reviewed (changes requested):
Super cool Mirha ☀️ The code is so much cleaner with these changes!

I've flagged some weird data mismatch in the Markdown entries. I don't know if I caught them all, I browsed the content directory.

If you want, we can do a pairing session to go through all of them and make sure it's all good.

mirhamasala commented on src/app/events/[slug]/utils/generateStructuredData.ts:
```diff
@@ -46,7 +46,7 @@ function getLocation({
 
 export function generateStructuredData(data: Event): WithContext<SchemaEvent> {
-  const {
+export function generateStructuredData(data: Event): WithContext<EventSchema> {
```
@CharlyMartin - Make sure when using suggestions to not add/remove unwanted things so that when we commit the suggestion the code doesn't break. 🙏🏼

CharlyMartin replied:
Good catch, thanks!

mirhamasala commented on src/app/_components/FeaturedEcosystemProjects.tsx:
```diff
@@ -13,7 +13,7 @@ import { PATHS } from '@/constants/paths'
 type FeaturedEcosystemProjectsProps = {
   ecosystemProjects: EcosystemProject[]
-}
+  ecosystemProjects: Array<EcosystemProject>
```
@CharlyMartin Same [^^](https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/494/commits/d49590caf370b5ee540f591d8c050bf1cde44df7#r1728205204)

mirhamasala commented on src/app/_components/FeaturedGrantGraduates.tsx:
```diff
@@ -13,7 +13,7 @@ import { PATHS } from '@/constants/paths'
 type FeaturedGrantsGraduatesProps = {
   grantGraduates: EcosystemProject[]
-}
+  grantGraduates: Array<EcosystemProject>
```
@CharlyMartin Same [^^](https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/494/commits/d49590caf370b5ee540f591d8c050bf1cde44df7#r1728205204)

mirhamasala commented:
>Super cool Mirha ☀️ The code is so much cleaner with these changes!

I've flagged some weird data mismatches in the Markdown entries. I don't know if I caught them all, I browsed the content directory. If you want, we can do a pairing session to go through all of them and make sure it's all good.

I've also added a few suggestions, all nits.

Thank you for the review, @CharlyMartin ~ I added Zod's `url()` for the URL fields in the schema and did another search and replace round (for `null` in all dynamic entries and `twitter` for the Ecosystem Explorer projects.
